blog/2025-06-12-oauth-improvements.md

```diff
@@ -0,0 +1,65 @@
+
+## Session Lifetimes
+
+Many developers are encountering friction with OAuth session lifetimes, especially using in-browser "public" clients. The existing lifetime limitations mean that un-refreshed sessions would time out after just two days.
+
+There will always be shorter session lifetimes for public clients, but we think the session lifetimes can be increased without a significant security trade-off. We are planning to roll out the following session lifetimes:
+
+- Public Clients  
+  - increase overall session lifetime (after which tokens can not be refreshed) from one week to two weeks  
+  - increase the lifetime of individual refresh tokens from two days to two weeks (the same as the overall session lifetime; refreshing tokens does not extend the overall session)  
+- Confidential Clients  
+  - increase the lifetime of individual refresh tokens from one month to three months (if tokens are not refreshed, the session ends before the overall session lifetime limit)  
+  - increase overall session lifetime (assuming tokens are refreshed) from one year to two years; including existing sessions
+
+Remember that the maximum lifetime of OAuth *access tokens* is much shorter: the specification recommends a limit of 30 minutes, and the reference implementation uses 15 minutes.
+
+There is a tracking issue here: [https://github.com/bluesky-social/atproto/pull/3883](https://github.com/bluesky-social/atproto/pull/3883) 
+
+## Auth Scopes
+
+Protocol design on Auth Scopes is wrapping up. This functionality will allow app developers to request granular permissions to specific atproto resources, like records (by NSID), remote API endpoints (using service proxying), account hosting status, and identity updates. The system will also allow Lexicon designers to define higher-level "bundles" of permissions for atproto resources in the same NSID namespace, to make end-user permission requests simpler.
+
+You can learn more by reading an earlier [draft of this protocol feature from March 2025](https://github.com/bluesky-social/atproto/discussions/3655).
+
+We will begin server-side implementation and integration work on this feature soon after a final proposal is published.
+
+## Other Changes and Bug Fixes
+
+The DPoP JWTs created by the `@atproto/oauth-client` TypeScript package incorrectly included query parameters as part of the `htu` request URL field, which goes against the DPoP specification. The PDS reference implementation was (and still is) permissive and would ignore this part of the URL. The client package has been fixed as of version [0.3.18](https://www.npmjs.com/package/@atproto/oauth-client?activeTab=versions).
```

Comment:
See my suggested "deprecation notice" bellow to understand why I suggest removing this phrase.

```suggestion
The DPoP JWTs created by the `@atproto/oauth-client` TypeScript package incorrectly included query parameters as part of the `htu` request URL field, which goes against the DPoP specification. The client package has been fixed as of version [0.3.18](https://www.npmjs.com/package/@atproto/oauth-client?activeTab=versions).
```